hazelcast/src/main/java/com/hazelcast/jet/impl/util/Util.java

```diff
@@ -140,6 +146,22 @@ public final class Util {
     private Util() {
     }
 
+    public static boolean checkJobIsAllowedToBeSuspended(TerminationMode terminationMode, JobConfig jobConfig) {
+        if (isJobSuspendable(jobConfig)) {
+            return true;
+        }
+
+        return !(terminationMode == SUSPEND_GRACEFUL
```

Comment:
how about `CANCEL_GRACEFUL`?

Reply:
It should not be present here, cancelation != suspension

Reply:
CANCEL_GRACEFUL creates snapshot or am I wrong? what is the use of such snapshot?

Reply:
Yes, it creates a snapshot. Since the last time we discussed it, we can technically export snapshots but we can't restore from it. Do we want to disallow snapshot export too?

Reply:
I think that current implementation is sufficient. `CANCEL_GRACEFUL` will not be used with named snapshots (EE implementation) for non-suspendable jobs. But even if it were used for graceful termination, it would only create a regular snapshot which is allowed for non-suspendable jobs.